proto/destination.proto

```diff
@@ -141,6 +149,29 @@ message Route {
   repeated ResponseClass response_classes = 2;
   // Metric labels to attach to requests and responses that match this route.
   map<string, string> metrics_labels = 3;
+  // If a route is retryable, any failed requests on that route may be retried
+  // by the proxy.
+  bool is_retryable = 4;
+  // After this time has elapsed since receiving the initial request, no more
+  // retries will be attempted.
+  google.protobuf.Duration retry_timeout = 5;
+}
+
+message RetryBudget {
```

Comment:
The retry budget implementation has some [assertions taken from finagle](https://github.com/tower-rs/tower/blob/master/tower-retry/src/budget.rs#L44-L49), should we document them here, and what happens if they aren't met?

Reply:
Good question.  Hopefully the TTL needing to be positive and the minRetriesPerSecond needing to be non-negative are obvious constraints.

I've formulated retry_ratio as a ratio here instead of as a percent like Finagle does because I feel like it's more clear.  The maximum retry percent of 1000% from Finagle would correspond to a retry ratio of 10.  This limit is pretty arbitrary and basically just a foot gun protection to make sure the value here isn't totally insane.  Not sure if we want those types of protections in place here, or at a higher layer (i.e. user facing configuration).